app/models/site_upload.rb

```diff
@@ -57,44 +48,10 @@ def assign_attributes(attributes)
     super(attributes.slice(:team).merge(attributes))
   end
 
-  def count
-    (new_sites&.length || 0) + (existing_sites&.length || 0)
-  end
-
-  def parse_sites
-    require "csv"
-
-    CSV.foreach(file.path, headers: true, encoding: "bom|utf-8", col_sep:) do |row|
-      row = row.to_h.transform_keys { |header| header.to_s.downcase } # Case-insensitive headers
-
-      url = Link.normalize(row["url"])
-      name = row["nom"] || row["name"]
-      tag_names = row["tags"].present? ? row["tags"].split(",").map(&:strip).compact_blank.uniq : []
-
-      row_tag_ids = tag_names.map { |n| team.tags.find_or_create_by(name: n).id }
-      combined_tag_ids = (tag_ids + row_tag_ids).uniq
-      existing_site = team.sites.find_by_url(url:)
-
-      if existing_site
-        existing_site.assign_attributes(tag_ids: combined_tag_ids.union(existing_site.tag_ids))
-        existing_site.assign_attributes(name:) unless existing_site.name
-        self.existing_sites[url] = existing_site
-      else
-        self.new_sites[url] = { url:, team:, name:, tag_ids: combined_tag_ids }
-      end
-    end
-  end
-
   private
 
-  def first_line
-    @first_line ||= File.open(file.path, &:gets)&.strip&.sub(BOM, "") || ""
-  end
-
-  def col_sep
-    SUPPORTED_SEPARATORS.max_by { |sep| first_line.count(sep) }
-  rescue StandardError
-    SUPPORTED_SEPARATORS.first
+  def parser
```

Comment:
```suggestion
  def parsed
```

Techniquement, c'est pas un parseur, c'est le résultat du parsing du fichier. Si on renomme la fonction,  on peut écrire `sites_data = parsed.data` et `missing_headers = REQUIRED_HEADERS - parsed.headers`, qui est plus parlant. On peut aussi mettre `parsed_file` si on veut que la méthode soit plus explicite hors contexte.